autorest/adal/token.go

```diff
@@ -850,11 +852,28 @@ func (spt *ServicePrincipalToken) getGrantType() string {
 }
 
 func isIMDS(u url.URL) bool {
-	imds, err := url.Parse(msiEndpoint)
+	return isMSIEndpoint(u) == true || isASEEndpoint(u) == true
```

Comment:
I think the requests for IMDS and ASE differ, don't they?

Reply:
The endpoints differ yes, or did you mean something else?

Reply:
And really, this function is poorly named.  It was created before we added support for ASE.

Reply:
Yes the function name is misleading. And yes I was wondering if those two MSI cases should be used interchangeably.

Reply:
I checked track2, maybe it's not essential for the requests in these cases but the headers do differ between IMDS and ASE scenarios.